clients/client-python/gravitino/api/expressions/named_reference.py

```diff
@@ -0,0 +1,76 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+from typing import List
+from gravitino.api.expressions.expression import Expression
+
+
+class NamedReference(Expression):
+    """
+    Represents a field or column reference in the public logical expression API.
+    """
+
+    @staticmethod
+    def field(field_name: List[str]) -> FieldReference:
+        """Returns a FieldReference for the given field name(s)."""
+        return FieldReference(field_name)
+
+    @staticmethod
+    def field_from_column(column_name: str) -> FieldReference:
+        """Returns a FieldReference for the given column name."""
+        return FieldReference([column_name])
+
+    def field_name(self) -> List[str]:
+        """
+        Returns the referenced field name as a list of string parts.
+        Must be implemented by subclasses.
+        """
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def children(self) -> List[Expression]:
+        """Named references do not have children."""
+        return Expression.EMPTY_EXPRESSION
+
+    def references(self) -> List[NamedReference]:
+        """Named references reference themselves."""
+        return [self]
+
+
+class FieldReference(NamedReference):
+    """
+    A NamedReference that references a field or column.
+    """
+
+    def __init__(self, field_name: List[str]):
+        super().__init__()
+        self._field_name = field_name
+
+    def field_name(self) -> List[str]:
```

Comment:
```suggestion
    def field_names(self) -> List[str]:
```

Reply:
It is a good suggestion. Just updated the `field_name` to `field_names`.

Reply:
Not resolved?